src/Pos/Client/Txp/History.hs

```diff
@@ -237,15 +236,16 @@ getBlockHistoryDefault
     :: forall ssc ctx m. TxHistoryEnv' ssc ctx m
     => [Address] -> m (DList TxHistoryEntry)
 getBlockHistoryDefault addrs = do
-    bot <- GS.getBot
-    sd <- GS.getSlottingData
-    systemStart <- getSystemStart
+
+    systemStart <- getSystemStartM
+    bot         <- GS.getBot
+    sd          <- GS.getSlottingData
```

Comment:
@pva701 we definitely need to store `SlottingData` in GState. We can store it using multiple keys (which is easy because this type is a map) or we can store the whole map by one key.
Initially it was implemented by using multiple keys: https://github.com/input-output-hk/cardano-sl/pull/1107/files#diff-495f1c9e499bcfb0abb523efc4bb09b2
Then this logic disappeared for unknown reason (as I asked in https://github.com/input-output-hk/cardano-sl/pull/1217#discussion_r130887026).
However, now I think that we probably can store this whole `SlottingData` in one key. We have 1 epoch per 5 days. It's only 71 epoch in a year. One `EpochSlottingData` takes tens of bytes (let's say at most 30, just 2 numbers). So it's only 2 Kb per year. Even in 10 years it will be only 20 Kb which is probably quite cheap (recall that we also store this data in memory, so we rarely read it from DB).